tests/functional/common.js

```diff
@@ -103,6 +103,18 @@ export function execWebExt(
   spawnOptions: child_process$spawnOpts,
 ): RunningWebExt {
 
+  if (spawnOptions.env) {
+    spawnOptions.env = {
+      // Propagate the current environment when redifining it from the spawnOptions
+      // otherwise it may trigger unexpected failures due to missing variables that
+      // are expected to be part of the environment (e.g. #2444 was failing only on
+      // windows because of this, due @pnpm/npm-conf, a transitive dependencies for
+      // update-notifier, expecting process.env.APPDATA to be defined when running
+      // on windows).
+      ...process.env,
+      ...spawnOptions.env,
+    };
+  }
```

Comment:
@willdurand I've been finally able to take a look into this, I was able to reproduce it consistently on a local windows system and in the inline comment here I'm trying to point out why we are doing this and what was going on in this particular case.

Let me know wdyt (included any opinion you may have about the clarity of this comment, and/or any typo to fix or rewording that could help to reduce it a bit but still leave enough details to recall why we are doing this here).

@pnpm/npm-conf is a transitive dependency for update-notifier, based on `npm why` it got in through the update-notifier direct dependency being updated in v6.0.2:
```
❯ npx npm@latest why node_modules/@pnpm/npm-conf
npx: installed 201 in 4.107s

@pnpm/npm-conf@1.0.4
node_modules/@pnpm/npm-conf
  @pnpm/npm-conf@"^1.0.4" from registry-auth-token@5.0.1
  node_modules/registry-auth-token
    registry-auth-token@"^5.0.1" from package-json@8.1.0
    node_modules/package-json
      package-json@"^8.1.0" from latest-version@7.0.0
      node_modules/latest-version
        latest-version@"^7.0.0" from update-notifier@6.0.2
        node_modules/update-notifier
          update-notifier@"6.0.2" from the root project
```

and it started to trigger this issue because @pnpm/npm-conf internally expects process.env.APPDATA to be defined while running on windows, and @pnpm/npm-conf was in the middle of node code executed implicitly as part of nodejs module loading.

As a side note, personally I find it quite annoying that update-notifier does somehow (likely indirectly) trigger this @pnpm/npm-conf module to be executed implicitly as part of some nodejs module loading hook, but I didn't dig more into it (at least not yet).